core/src/main/java/com/scalar/db/api/ConditionalExpression.java

```diff
@@ -19,137 +19,266 @@
+  public ConditionalExpression(String columnName, String textValue, Operator operator) {
+    this(columnName, new TextValue(textValue), operator);
   }
 
   /**
-   * Constructs a {@code ConditionalExpression} with the specified name, value and operator.
+   * Constructs a {@code ConditionalExpression} with the specified column name, value and operator.
    *
-   * @param name name of target value
-   * @param value value used to compare with the target value
-   * @param operator operator used to compare the target value specified with the name and the value
+   * @param columnName a name of target column
+   * @param blobValue a BLOB value used to compare with the target column
+   * @param operator an operator used to compare the target column
    */
-  public ConditionalExpression(String name, byte[] value, Operator operator) {
-    this(name, new BlobValue(value), operator);
+  public ConditionalExpression(String columnName, byte[] blobValue, Operator operator) {
+    this(columnName, new BlobValue(blobValue), operator);
   }
 
   /**
-   * Constructs a {@code ConditionalExpression} with the specified name, value and operator.
+   * Constructs a {@code ConditionalExpression} with the specified column name, value and operator.
    *
-   * @param name name of target value
-   * @param value value used to compare with the target value
-   * @param operator operator used to compare the target value specified with the name and the value
+   * @param columnName a name of target column
+   * @param blobValue a BLOB value used to compare with the target column
+   * @param operator an operator used to compare the target column
    */
-  public ConditionalExpression(String name, ByteBuffer value, Operator operator) {
-    this(name, new BlobValue(value), operator);
+  public ConditionalExpression(String columnName, ByteBuffer blobValue, Operator operator) {
+    this(columnName, new BlobValue(blobValue), operator);
   }
 
   /**
-   * Returns the name of target value
+   * Returns the column name of target column.
    *
-   * @return the name of target value
+   * @return the column name of target column
+   * @deprecated As of release 3.6.0. Will be removed in release 5.0.0
    */
+  @Deprecated
   public String getName() {
-    return name;
+    return columnName;
+  }
+
+  /**
+   * Returns the column name of target column.
+   *
+   * @return the column name of target column
+   */
+  public String getColumnName() {
+    return columnName;
   }
 
   /**
-   * Return the value used to compare with the target value
+   * Return the value used to compare with the target column.
    *
-   * @return the value used to compare with the target value
+   * <p>This method is primarily for internal use. Breaking changes can and will be introduced to
+   * this method. Users should not depend on it.
+   *
+   * @return the value used to compare with the target column
    */
   public Value<?> getValue() {
     return value;
   }
```

Comment:
Also, I made this `getValue()` method internal use. I just documented it in Javadoc.